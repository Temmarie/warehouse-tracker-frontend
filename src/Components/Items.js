import React, { useEffect } from 'react';
import { Link } from 'react-router-dom';
import { useSelector, useDispatch } from 'react-redux';
import Nav from './Nav';
import { fetchItems } from '../Actions';

const Items = () => {
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(fetchItems());
  }, []);

  const itemData = useSelector((state) => state.itemReducer);
  const user = useSelector((state) => state.userReducer);

  return (
    <>
      <div className="App">
        <Nav />
        <span>
          Hey
          {user.username}
          , id:
          {user.id}
        </span>
        <div>
          {itemData.map((item) => (
            <p key={item.code}>{item.name}</p>
          ))}
        </div>
        <div className="navbar-item">
          <Link to="/">Back</Link>
        </div>
      </div>
    </>
  );
};

// const selector = (state) => ({
//   user: state.userReducer,
//   itemData: state.itemReducer,
// })
// export default connect(selector, null)(Items);

export default Items;
