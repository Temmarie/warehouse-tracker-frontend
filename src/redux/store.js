import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import logger from 'redux-logger';
import registrationReducer from './user/registrationReducer';

const store = createStore(registrationReducer, applyMiddleware(thunk, logger));

export default store;
