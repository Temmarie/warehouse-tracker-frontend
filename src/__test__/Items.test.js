import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import Items from '../Components/Items';
import store from '../Reducers/index';

describe('Items component', () => {
  it('Matches the snapshot', () => {
    const tree = renderer.create(<Provider store={store}><Items /></Provider>).toJSON;
    expect(tree).toMatchSnapshot();
  });
});
