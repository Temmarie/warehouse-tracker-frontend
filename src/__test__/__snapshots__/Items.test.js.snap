// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Items component Matches the snapshot 1`] = `[Function]`;

exports[`RegisterContainer component Matches the snapshot 1`] = `[Function]`;
